web/src/user_profile.js

```diff
@@ -284,6 +355,45 @@ export function register_click_handlers() {
         e.preventDefault();
     });
 
+    $("body").on("click", "#user-profile-modal .add-subscription-button", (e) => {
+        e.preventDefault();
+        const stream_id = Number.parseInt(user_profile_subscribe_widget.value(), 10);
+        const sub = sub_store.get(stream_id);
+        const $alert_box = $("#user-profile-streams-tab .stream_list_info");
+        function addition_success(data) {
+            if (Object.keys(data.subscribed).length > 0) {
+                ui_report.success(
+                    $t_html({defaultMessage: "subscribed successfully!"}),
+                    $alert_box,
+                    1200,
+                );
+            } else {
+                ui_report.client_error(
+                    $t_html({defaultMessage: "Already subscribed."}),
+                    $alert_box,
+                    1200,
```

Comment:
How did you pick these 1200s?

Reply:
We use 1200 for most of the alerts and banners in Zulip. Additionally, the "remove subscription" button also had a 1200s removal time. To maintain consistency with this pattern, I set the time to 1200. If you have anything else in your mind, do let me know.